### PR TITLE
Make proxy PRIVATE in verbose

### PR DESCRIPTION
**Pull request by bato3**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
now:
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--proxy', u'PRIVATE', u'http://www.crunchyroll.com/media-723735', u'-v']
[debug] Encodings: locale cp1250, fs mbcs, out cp852, pref cp1250
[debug] youtube-dl version 2018.07.21
[debug] Git HEAD: be477061e
[debug] Python version 2.7.14 (CPython) - Windows-10-10.0.17134
[debug] exe versions: ffmpeg 4.0, ffprobe 4.0, phantomjs 2.1.1
[debug] Proxy map: {u'http': u'http://PRIVATE@uk.vpn.mydevil.net:8889', u'https': u'http://PRIVATE@uk.vpn.mydevil.net:8889'}
```

### CODE REVIEW

File: youtube_dl/YoutubeDL.py [hunk 1]
```diff
@@ -2273,7 +2273,10 @@ def python_implementation():
         for handler in self._opener.handlers:
             if hasattr(handler, 'proxies'):
                 proxy_map.update(handler.proxies)
-        self._write_string('[debug] Proxy map: ' + compat_str(proxy_map) + '\n')
+        self._write_string('[debug] Proxy map: ' + re.sub(
+            r'(https?://)(?:(?:[^:/\'"@]+:)?[^@]+)(@)',
```
Comment (dstftw) [inline youtube_dl/YoutubeDL.py]: 1. There are more schemes possible as well as missing scheme.
2. `'"` are allowed.

File: youtube_dl/options.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 
 def _hide_login_info(opts):
-    PRIVATE_OPTS = set(['-p', '--password', '-u', '--username', '--video-password', '--ap-password', '--ap-username'])
+    PRIVATE_OPTS = set(['-p', '--password', '-u', '--username', '--video-password', '--ap-password', '--ap-username', '--proxy'])
```
Comment (dstftw) [inline youtube_dl/options.py]: + `--geo-verification-proxy`.

**Comment by dstftw** [comment]
Inconsistency:
>u'--proxy', u'PRIVATE'
>u'http': u'http://PRIVATE@uk.vpn.mydevil.net:8889'

**Comment by bato3** [comment]
@dstftw 
>Inconsistency:
>
>>    u'--proxy', u'PRIVATE'
>>    u'http': u'http://PRIVATE@uk.vpn.mydevil.net:8889'

I would like to delete the entire `Proxy map` section, but for some reason it has been added...
For `--proxy` I use native function. It makes sense only delete login data?

**Comment by dstftw** [comment]
1. Don't remove proxy map.
2. For command lines proxies it should behave similarly to proxy map - only hiding `userinfo` part.

File: youtube_dl/options.py [hunk 3]
```diff
@@ -20,21 +20,37 @@
...
+                return m.group('key') + '=' + hide_url_login_info(m.group('url'))
+            else:
+                return o
 
     opts = list(map(_scrub_eq, opts))
     for idx, opt in enumerate(opts):
         if opt in PRIVATE_OPTS and idx + 1 < len(opts):
             opts[idx + 1] = 'PRIVATE'
+        else:
+            if opt in URL_PRIVATE_OPTS and idx + 1 < len(opts):
```
Comment (dstftw) [inline youtube_dl/options.py]: elif

File: youtube_dl/options.py [hunk 4]
```diff
@@ -20,21 +20,37 @@
 from .version import __version__
 
 
+def hide_url_login_info(url):
+    return re.sub(
+        r'((?:http|socks5)s?://)(?:(?:[^:@]+:)?[^@]+)(@)',
```
Comment (dstftw) [inline youtube_dl/options.py]: Still more protocols possible. Still no schemeless support.

**Pull request closed without merging**